File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    int Count(ListNode*head){
        ListNode* temp = head;
        int i=1;
        while(temp->next){
            i++;
            temp=temp->next;
        }
        return i;

    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode*temp = head;
        if(!head || n==0){return head;}
        if(n==1){
            head=head->next;
            temp->next=nullptr;
            return head;
        }
        int len=Count(head);
        cout<<len<<endl;
        cout<<"dsf"<<endl;
        int ne=len-n;

        int i=1;
        if(ne>=0){
         while(i<ne){
            cout<<temp->val<<endl;
            temp=temp->next;
            cout<<"ith value"<<i<<endl;
            i++;
         }
        }
        ListNode* t1=temp->next;
        temp->next=t1->next;
        return head;
        
    }
};

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-1.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func hasCycle(head *ListNode) bool {
    //when using := this you don't need define *ListNode it infers automatically of the type
    current :=head
    fast :=current
    slow  :=current

    for fast!= nil && fast.Next!=nil{
        slow=slow.Next
        fast=fast.Next.Next
        if slow == fast {
            return true
        }

    } 
    return false;
}
